src/utils/preRestart.ts

```diff
@@ -1,28 +1,38 @@
 /* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { Signer } from 'ethers'
 import { Route } from '../types'
 import { LifiErrorCode } from './errors'
 
-export const handlePreRestart = (route: Route) => {
+export const handlePreRestart = (route: Route, signer: Signer) => {
   for (let index = 0; index < route.steps.length; index++) {
     const stepHasFailed = route.steps[index].execution?.status === 'FAILED'
 
     if (stepHasFailed) {
-      handleErrorType(route, index)
+      handleErrorType(route, index, signer)
       deleteFailedProcesses(route, index)
       deleteTransactionData(route, index)
     }
   }
 }
 
-const handleErrorType = (route: Route, index: number) => {
+const handleErrorType = (route: Route, index: number, signer: Signer) => {
   const isGasLimitError = route.steps[index].execution?.process.some(
     (p) => p.error?.code === LifiErrorCode.GasLimitError
   )
   const isGasPriceError = route.steps[index].execution?.process.some(
     (p) => p.error?.code === LifiErrorCode.TransactionUnderpriced
   )
 
+  const { transactionRequest } = route.steps[index]
+
   if (isGasLimitError) {
+    if (transactionRequest) {
+      const gasLimit = signer.estimateGas(transactionRequest)
```

Comment:
Can this potentially throw something?

Reply:
This ideally shouldn't throw any errors until there's an issue from RPC or an incorrect `transactionRequest` being sent to it.

But to be on safer side, wrapped in `try...catch` block and also handling the promises correctly now.